cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/chart.go

```diff
@@ -81,7 +95,22 @@ func (s *Server) availableChartDetail(ctx context.Context, repoName types.Namesp
 		return nil, err
 	}
 
-	return availablePackageDetailFromChartDetail(chartID, chartDetail)
+	pkgDetail, err := availablePackageDetailFromChartDetail(chartID, chartDetail)
+	if err != nil {
+		return nil, err
+	}
+
+	// fix up a couple of fields that don't come from the chart tarball
+	repoUrl := repo.Spec.URL
+	if repoUrl == "" {
+		return nil, status.Errorf(codes.NotFound, "Missing required field spec.url on repository %q", repoName)
```

Comment:
Hmmm, not sure about a not found here? With helm you can install a chart from a repo, then later delete the repo. Is that same situation impossible with flux? Or do we return a not found in helm?

Reply:
This was here a long time, I just moved the code around a bit. I think it was for some negative test I wrote. Anyway, `repo.Spec.URL` is a required field in flux https://fluxcd.io/docs/components/source/helmrepositories/ so in theory the if clause never triggers under normal circumstances

Reply:
regarding your question about fix. The helm chart CR has a back reference to the source. So I don't think this situation is possible. If you delete the helm repo, then associated helmcharts are deleted too. But I will verify now just to make sure

Reply:
I stand corrected. It is possible to delete the helmrepository. Then the helm chart goes into a 'bad' (not ready) state:
```
[v1alpha1]$ kubectl get helmchart
NAME                 CHART     VERSION   SOURCE KIND      SOURCE NAME   READY   STATUS                                         AGE
default-my-podinfo   podinfo   *         HelmRepository   podinfo       True    Pulled 'podinfo' chart with version '6.1.0'.   14s
[v1alpha1]$ kubectl get helmrepository
NAME      URL                                      READY   STATUS                                                                               AGE
podinfo   https://stefanprodan.github.io/podinfo   True    Fetched revision: 4995e6e5e300c68f9ea8e0f26c91e42f0dbaf1be46e4e464b25988eb5ae86d4b   97s
[v1alpha1]$ kubectl delete helmrepository/podinfo
helmrepository.source.toolkit.fluxcd.io "podinfo" deleted
[v1alpha1]$ kubectl get helmchart
NAME                 CHART     VERSION   SOURCE KIND      SOURCE NAME   READY   STATUS                                         AGE
default-my-podinfo   podinfo   *         HelmRepository   podinfo       True    Pulled 'podinfo' chart with version '6.1.0'.   56s
[v1alpha1]$ kubectl get helmchart
NAME                 CHART     VERSION   SOURCE KIND      SOURCE NAME   READY   STATUS                                                                                   AGE
default-my-podinfo   podinfo   *         HelmRepository   podinfo       False   failed to retrieve source: HelmRepository.source.toolkit.fluxcd.io "podinfo" not found   64s
[v1alpha1]$ kubectl get helmchart
NAME                 CHART     VERSION   SOURCE KIND      SOURCE NAME   READY   STATUS                                                                                   AGE
default-my-podinfo   podinfo   *         HelmRepository   podinfo       False   failed to retrieve source: HelmRepository.source.toolkit.fluxcd.io "podinfo" not found   69s
```
When this happens flux plugin `GetAvailablePackageDetail`() will raise a NotFound error a lot earier than the line you pointed out. It'll be on line 52 (repo, err := s.getRepoInCluster(ctx, repoName)) call. If you think we should change the way it works, let me know

Reply:
Yeah, fine as is then. Thanks for checking.